src/core/manager.c

```diff
@@ -3180,6 +3186,15 @@ int manager_deserialize(Manager *m, FILE *f, FDSet *fds) {
         return r;
 }
 
+static void manager_flush_finished_jobs(Manager *m) {
+        Job *j;
+
+        while ((j = set_steal_first(m->pending_finished_jobs))) {
+                bus_job_send_removed_signal(j);
+                job_free(j);
+        }
```

Comment:
...and also `pending_finished_jobs` should be freed here, no need to keep it around afterwards.